PlayTools/Controls/Backend/Action/PlayAction.swift

```diff
@@ -369,14 +368,14 @@ class SwipeAction: Action {
             return
         }
         Toucher.touchcam(point: self.location, phase: UITouch.Phase.ended, tid: &id)
+        timer.suspend()
         delay(0.02) {
             self.cooldown = false
         }
         cooldown = true
     }
 
     func invalidate() {
-        timer.suspend()
         self.doLiftOff()
```

Comment:
There is a low probability contention case, where `invalidate` and `checkEnded` are called from two threads (main thread and `touchQueue`), then they call `doLiftOff` simultaneously and suspend timer twice. 

I think a solution can be

```suggestion
        PlayInput.touchQueue.async(execute: self.doLiftOff)
```

and all invocations of `doLiftOff` would be from the same thread. (btw why is this method public lol)